rust_src/src/windows.rs

```diff
@@ -1967,4 +1967,30 @@ pub fn set_window_fringes_lisp(
     }
 }
 
+/// Return pixel width of window WINDOW before last size changes.
+/// WINDOW must be a valid window and defaults to the selected one.
+///
+/// The return value is the pixel width of WINDOW at the last time
+/// `window-size-change-functions' was run.  It's zero if WINDOW was made
+/// after that.
+#[lisp_fn(min = "0")]
+pub fn window_pixel_width_before_size_change(window: LispWindowValidOrSelected) -> EmacsInt {
+    LispWindowRef::from(window)
+        .pixel_width_before_size_change
+        .into()
+}
+
+/// Return pixel height of window WINDOW before last size changes.
+/// WINDOW must be a valid window and defaults to the selected one.
+///
+/// The return value is the pixel height of WINDOW at the last time
+/// `window-size-change-functions' was run.  It's zero if WINDOW was made
+/// after that.
+#[lisp_fn(min = "0")]
+pub fn window_pixel_height_before_size_change(window: LispWindowValidOrSelected) -> EmacsInt {
```

Comment:
You can return i32 in both functions and drop the `into`.

Reply:
Is there any difference between `i32` and `EmacsInt` (which is i64, iirc) for Remacs? I see that some functions (like `window_new_pixel`) return `EmacsInt`, and some `i32`. Won't there be a problem when negative will be returned?

Reply:
`EmacsInt` is a `i64`. I didn't take a look at `window_new_pixel`, but I think this was ported some time ago or it passed the review. `lisp_fn` casts the returned `i32` to a `LispObject`.

Reply:
Thx for the explanation. Fixed.